src/riak_test_escript.erl

```diff
@@ -21,299 +21,436 @@
+
+    ok = prepare(ParsedArgs, Tests, NonTests),
+    Results = execute(Tests, OutDir, ParsedArgs),
+    finalize(Results, ParsedArgs).
+
+% @doc Validate the command-line options
+parse_args(Args) ->
+    validate_args(getopt:parse(cli_options(), Args)).
+
+validate_args({ok, {[], _}}) ->
+    print_help();
+validate_args({ok, {ParsedArgs, NonOptionArgs}}) ->
+    case proplists:is_defined(help, ParsedArgs) of
+        true ->
+            print_help();
+        _ ->
+            {ParsedArgs, NonOptionArgs}
+    end;
+validate_args(_) ->
+    print_help().
 
-cli_options() ->
 %% Option Name, Short Code, Long Code, Argument Spec, Help Message
+cli_options() ->
 [
- {help,               $h, "help",     undefined,  "Print this usage page"},
- {config,             $c, "conf",     string,     "specifies the project configuration"},
- {tests,              $t, "tests",    string,     "specifies which tests to run"},
- {suites,             $s, "suites",   string,     "which suites to run"},
- {dir,                $d, "dir",      string,     "run all tests in the specified directory"},
- {skip,               $x, "skip",     string,     "list of tests to skip in a directory"},
- {verbose,            $v, "verbose",  undefined,  "verbose output"},
- {outdir,             $o, "outdir",   string,     "output directory"},
- {backend,            $b, "backend",  atom,       "backend to test [memory | bitcask | eleveldb]"},
- {upgrade_version,    $u, "upgrade",  atom,       "which version to upgrade from [ previous | legacy ]"},
- {keep,        undefined, "keep",     boolean,    "do not teardown cluster"},
- {report,             $r, "report",   string,     "you're reporting an official test run, provide platform info (e.g. ubuntu-1204-64)\nUse 'config' if you want to pull from ~/.riak_test.config"},
- {file,               $F, "file",     string,     "use the specified file instead of ~/.riak_test.config"}
+ {help,                   $h, "help",     undefined,  "Print this usage page"},
+ {config,                 $c, "conf",     string,     "specifies the project configuration"},
+ {tests,                  $t, "tests",    string,     "specifies which tests to run"},
+ {suites,                 $s, "suites",   string,     "which suites to run"},
+ {groups,                 $g, "groups",   string,     "specifiy a list of test groups to run"},
+ {dir,                    $d, "dir",      string,     "run all tests in the specified directory"},
+ {skip,                   $x, "skip",     string,     "list of tests to skip in a directory"},
+ {verbose,                $v, "verbose",  undefined,  "verbose output"},
+ {outdir,                 $o, "outdir",   string,     "output directory"},
+ {backend,                $b, "backend",  atom,       "backend to test [memory | bitcask | eleveldb]"},
+ {upgrade_path,           $u, "upgrade-path", atom,   "comma-separated list representing an upgrade path (e.g. riak-1.3.4,riak_ee-1.4.12,riak_ee-2.0.0)"},
```

Comment:
Yes. Same with the config options.  Still need to figure out how those will work.  We'll have to address it once we port an upgrade test.